File: ios/Packages/Contracts/Sources/Contracts/TenantConfig.swift
import Foundation

/// Tenant-specific theme configuration.
public struct TenantTheme: Equatable {
    public let primaryColor: String
    public let secondaryColor: String
    public let backgroundColor: String
    public let logoUrl: String?

    public init(
        primaryColor: String = "#6200EE",
        secondaryColor: String = "#03DAC6",
        backgroundColor: String = "#FFFFFF",
        logoUrl: String? = nil
    ) {
        self.primaryColor = primaryColor
        self.secondaryColor = secondaryColor
        self.backgroundColor = backgroundColor
        self.logoUrl = logoUrl
    }
}

/// Tenant-specific API configuration.
public struct ApiConfig: Equatable {
    public let baseUrl: String
    public let apiKey: String?
    public let timeout: TimeInterval

    public init(
        baseUrl: String = "https://api.example.com",
        apiKey: String? = nil,
        timeout: TimeInterval = 30
    ) {
        self.baseUrl = baseUrl
        self.apiKey = apiKey
        self.timeout = timeout
    }
}

/// Per-tenant configuration for multi-tenant / white-label support.
///
/// Mirrors Android `TenantConfig`. Delivered via `ModuleContext.tenantConfig`.
public struct TenantConfig: Equatable {
    public let tenantId: String
    public let displayName: String

    /// Module IDs enabled for this tenant. Empty = all modules enabled.
    public let enabledModules: [String]
    public let theme: TenantTheme
    public let apiConfig: ApiConfig

    public init(
        tenantId: String,
        displayName: String,
        enabledModules: [String] = [],
        theme: TenantTheme = TenantTheme(),
        apiConfig: ApiConfig = ApiConfig()
    ) {
        self.tenantId = tenantId
        self.displayName = displayName
        self.enabledModules = enabledModules
        self.theme = theme
        self.apiConfig = apiConfig
    }
}

File: flutter/contracts/lib/src/tenant_config.dart
/// Tenant-specific theme configuration.
class TenantTheme {
  final String primaryColor;
  final String secondaryColor;
  final String backgroundColor;
  final String? logoUrl;

  const TenantTheme({
    this.primaryColor = '#6200EE',
    this.secondaryColor = '#03DAC6',
    this.backgroundColor = '#FFFFFF',
    this.logoUrl,
  });
}

/// Tenant-specific API configuration.
class ApiConfig {
  final String baseUrl;
  final String? apiKey;
  final int timeout;

  const ApiConfig({
    this.baseUrl = 'https://api.example.com',
    this.apiKey,
    this.timeout = 30000,
  });
}

/// Per-tenant configuration for multi-tenant / white-label support.
///
/// Mirrors Android [TenantConfig]. Provided via [ModuleContext.tenantConfig].
class TenantConfig {
  final String tenantId;
  final String displayName;

  /// Module IDs enabled for this tenant. Empty = all modules enabled.
  final List<String> enabledModules;
  final TenantTheme theme;
  final ApiConfig apiConfig;

  const TenantConfig({
    required this.tenantId,
    required this.displayName,
    this.enabledModules = const [],
    this.theme = const TenantTheme(),
    this.apiConfig = const ApiConfig(),
  });
}
